### ISSUE TITLE
Backups stuck "inProgress" indefinitely 

### ISSUE THREAD

**Issue by yogeek** [vmware-tanzu/velero#4003]
We use velero 1.3.2 (installed with the CLI) and some backups are staying `In Progress` indefinitely. 
These backups are launched by a schedule but the problem is also present with manual backup (only done for testing this issue) :

<details>
  <summary>Backup details (click to expand)</summary>
```

	  $ velero schedule get
	  NAME            STATUS    CREATED                          SCHEDULE     BACKUP TTL   LAST BACKUP   SELECTOR
	  daily-backup    Enabled   2021-07-28 19:04:44 +0200 CEST   0 23 * * *   168h0m0s     1h ago        <none>
	  hourly-backup   Enabled   2021-07-28 19:04:43 +0200 CEST   0 * * * *    24h0m0s      56m ago       <none>
	  weekly-backup   Enabled   2021-07-28 19:04:45 +0200 CEST   0 23 * * 0   720h0m0s     1h ago        <none>

	  
	  $ velero backup create test-backup --snapshot-volumes=false
	  Backup request "test-backup" submitted successfully.
	  Run `velero backup describe test-backup` or `velero backup logs test-backup` for more details.

	  $ velero backup get
		    NAME                           STATUS       CREATED                          EXPIRES   STORAGE LOCATION   SELECTOR
	  daily-backup-20210728170444    InProgress   2021-07-28 19:05:36 +0200 CEST   6d        default            <none>
	  daily-backup-20210726230018    Completed    2021-07-27 01:00:18 +0200 CEST   5d        default            <none>
	  hourly-backup-20210728190032   InProgress   2021-07-28 21:00:32 +0200 CEST   23h       default            <none>
	  hourly-backup-20210728180013   InProgress   2021-07-28 20:00:13 +0200 CEST   22h       default            <none>
	  hourly-backup-20210728170443   InProgress   2021-07-28 19:04:43 +0200 CEST   21h       default            <none>
	  hourly-backup-20210728160039   Completed    2021-07-28 18:00:39 +0200 CEST   20h       default            <none>
	  hourly-backup-20210728120024   Completed    2021-07-28 14:00:24 +0200 CEST   16h       default            <none>
	  hourly-backup-20210728100055   Completed    2021-07-28 12:00:55 +0200 CEST   14h       default            <none>
	  hourly-backup-20210728080030   Completed    2021-07-28 10:00:30 +0200 CEST   12h       default            <none>
	  hourly-backup-20210728040040   Completed    2021-07-28 06:00:40 +0200 CEST   8h        default            <none>
	  hourly-backup-20210728020016   Completed    2021-07-28 04:00:16 +0200 CEST   6h        default            <none>
	  hourly-backup-20210727230005   Completed    2021-07-28 01:00:28 +0200 CEST   3h        default            <none>
	  hourly-backup-20210727220005   Completed    2021-07-28 00:00:05 +0200 CEST   2h        default            <none>
	  test-backup                    InProgress   2021-07-28 20:58:34 +0200 CEST   29d       default            <none>
	  weekly-backup-20210728170445   InProgress   2021-07-28 19:06:50 +0200 CEST   29d       default            <none>
	  weekly-backup-20210704230010   Completed    2021-07-05 01:01:17 +0200 CEST   6d        default            <none>

	  ```
</details>

When I did a describe on this backups, I do not see any more details...

<details>
<summary>Describe details (click to expand)</summary>

	```
	  Name:         test-backup
	  Namespace:    velero
	  Labels:       velero.io/storage-location=default
	  Annotations:  <none>
	  
	  Phase:  InProgress
	  
	  Namespaces:
	    Included:  *
	    Excluded:  <none>
	  
	  Resources:
	    Included:        *
	    Excluded:        <none>
	    Cluster-scoped:  auto
	  
	  Label selector:  <none>
	  
	  Storage Location:  default
	  
	  Snapshot PVs:  false
	  
	  TTL:  720h0m0s
	  
	  Hooks:  <none>
	  
	  Backup Format Version:  1
	  
	  Started:    2021-07-28 20:58:34 +0200 CEST
	  Completed:  <n/a>
	  
	  Expiration:  2021-08-27 20:58:34 +0200 CEST
	  
	  Persistent Volumes: <none included>

	```
</details>

I tried to update in the 1.4.3 version, but still the same issue...

I tried to uninstall/install velero 1.3.2 again but the issue is still there.

and here are the logs about one InProgressbackup I had before reinstalling :

```
$ kubectl logs velero-596bf8b4d8-qzlb4 | grep hourly-backup-20210706110055 
time="2021-07-06T12:03:20Z" level=info msg="Processing item" backup=velero/hourly-backup-20210706120027 logSource="pkg/backup/backup.go:371" name=hourly-backup-20210706110055 namespace=velero progress= resource=backups.velero.io
time="2021-07-06T12:03:20Z" level=info msg="Backing up item" backup=velero/hourly-backup-20210706120027 logSource="pkg/backup/item_backupper.go:120" name=hourly-backup-20210706110055 namespace=velero resource=backups.velero.io
time="2021-07-06T12:03:20Z" level=info msg="Backed up 11857 items out of an estimated total of 12380 (estimate will change throughout the backup)" backup=velero/hourly-backup-20210706120027 logSource="pkg/backup/backup.go:411" name=hourly-backup-20210706110055 namespace=velero progress= resource=backups.velero.

```

The pod seems to be restarted frequently :
```
kgp                                      
NAME                      READY   STATUS    RESTARTS   AGE
velero-79499f49dc-lt58d   1/1     Running   6          120m
```
but looking at the resource consumption I do not see anything weird...

I do no know if it is relevant by the cluster has more than 30 nodes (average because autoscaler) and around 1000 pods (average)

```
velero backup logs test-backup                          
Logs for backup "test-backup" are not available until it's finished processing. Please wait until the backup has a phase of Completed or Failed and try again.
```

**Anything else you would like to add:**
We have several clusters and the problem seems to be only on clusters with more nodes (>30) and pods (>1000) (our smaller cluster does not have the issue with the same velero version)

**Environment:**

- Velero version (use `velero version`): 1.3.2 but but had the same issue with 1.4.3 
```
velero version                    
Client:
	Version: v1.3.2
	Git commit: 4fc1c700282d6ece5b3ffc03cf296c435ccc903e
Server:
	Version: v1.3.2
```

- Velero features (use `velero client config get features`): `features: <NOT SET>`
- Kubernetes version (use `kubectl version`): 1.19.8
- Kubernetes installer & version: kubeadm on AWS EC2
- Cloud provider or hardware configuration: AWS
- OS (e.g. from `/etc/os-release`): Ubuntu 18 EC2 instances

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by dsu-igeek** [comment]
Current release is 1.6.2.  Please upgrade to that and see if the problem persists.

**Comment by echel0n** [comment]
I'm having a similar issue on v1.6.2, stuck at inProgress, this was working before and I've made no config changes.

velero backup logs velero-daily-backup-20210802201850
`Logs for backup "velero-daily-backup-20210802201850" are not available until it's finished processing. Please wait until the backup has a phase of Completed or Failed and try again.`

kubectl logs -n velero velero-5595dd69c-j8wvx
`I0802 20:50:14.471441       1 request.go:645] Throttling request took 1.039441687s, request: GET:https://10.96.0.1:443/apis/rabbitmq.com/v1beta1?timeout=32s
time="2021-08-02T20:50:22Z" level=info msg="Validating backup storage location" backup-storage-location=default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:114"
time="2021-08-02T20:50:22Z" level=info msg="Backup storage location valid, marking as available" backup-storage-location=default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:121"
time="2021-08-02T20:51:22Z" level=info msg="Validating backup storage location" backup-storage-location=default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:114"
time="2021-08-02T20:51:22Z" level=info msg="Backup storage location valid, marking as available" backup-storage-location=default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:121"`

**Comment by yogeek** [comment]
@dsu-igeek I updated and the problem persists :
```
$ velero version                        
Client:
	Version: v1.6.2
	Git commit: 8c9cdb9603446760452979dc77f93b17054ea1cc
Server:
	Version: v1.6.2

$ velero backup get                     
NAME                           STATUS       ERRORS   WARNINGS   CREATED                          EXPIRES   STORAGE LOCATION   SELECTOR
daily-backup-20210802230050    InProgress   0        0          2021-08-03 01:00:50 +0200 CEST   6d        default            <none>
daily-backup-20210801230013    InProgress   0        0          2021-08-02 01:00:13 +0200 CEST   5d        default            <none>
daily-backup-20210731230036    InProgress   0        0          2021-08-01 01:00:36 +0200 CEST   4d        default            <none>
daily-backup-20210730230040    InProgress   0        0          2021-07-31 01:01:17 +0200 CEST   3d        default            <none>
daily-backup-20210729230019    InProgress   0        0          2021-07-30 01:00:19 +0200 CEST   2d        default            <none>
daily-backup-20210728230005    Completed    0        0          2021-07-29 01:00:05 +0200 CEST   1d        default            <none>
daily-backup-20210728170444    InProgress   0        0          2021-07-28 19:05:36 +0200 CEST   1d        default            <none>
hourly-backup-20210803080007   InProgress   0        0          2021-08-03 10:00:07 +0200 CEST   23h       default            <none>
hourly-backup-20210803070021   InProgress   0        0          2021-08-03 09:00:21 +0200 CEST   22h       default            <none>
hourly-backup-20210803060024   InProgress   0        0          2021-08-03 08:00:24 +0200 CEST   21h       default            <none>
hourly-backup-20210803050051   InProgress   0        0          2021-08-03 07:00:51 +0200 CEST   20h       default            <none>
hourly-backup-20210803040020   InProgress   0        0          2021-08-03 06:00:20 +0200 CEST   19h       default            <none>
hourly-backup-20210803030050   InProgress   0        0          2021-08-03 05:00:50 +0200 CEST   18h       default            <none>
hourly-backup-20210803020021   InProgress   0        0          2021-08-03 04:00:21 +0200 CEST   17h       default            <none>
hourly-backup-20210803010046   InProgress   0        0          2021-08-03 03:00:46 +0200 CEST   16h       default            <none>
hourly-backup-20210803000012   InProgress   0        0          2021-08-03 02:00:12 +0200 CEST   15h       default            <none>
hourly-backup-20210802230050   InProgress   0        0          2021-08-03 01:01:29 +0200 CEST   14h       default            <none>
hourly-backup-20210802220014   InProgress   0        0          2021-08-03 00:00:15 +0200 CEST   13h       default            <none>
hourly-backup-20210802210043   InProgress   0        0          2021-08-02 23:00:43 +0200 CEST   12h       default            <none>
hourly-backup-20210802200011   InProgress   0        0          2021-08-02 22:00:11 +0200 CEST   11h       default            <none>
hourly-backup-20210802190039   InProgress   0        0          2021-08-02 21:00:39 +0200 CEST   10h       default            <none>
hourly-backup-20210802180008   InProgress   0        0          2021-08-02 20:00:08 +0200 CEST   9h        default            <none>
hourly-backup-20210802170039   InProgress   0        0          2021-08-02 19:00:39 +0200 CEST   8h        default            <none>
hourly-backup-20210802160004   InProgress   0        0          2021-08-02 18:00:04 +0200 CEST   7h        default            <none>
hourly-backup-20210802150036   InProgress   0        0          2021-08-02 17:00:36 +0200 CEST   6h        default            <none>
hourly-backup-20210802140008   InProgress   0        0          2021-08-02 16:00:08 +0200 CEST   5h        default            <none>
hourly-backup-20210802130038   InProgress   0        0          2021-08-02 15:00:38 +0200 CEST   4h        default            <none>
hourly-backup-20210802120011   InProgress   0        0          2021-08-02 14:00:11 +0200 CEST   3h        default            <none>
hourly-backup-20210802110039   InProgress   0        0          2021-08-02 13:00:39 +0200 CEST   2h        default            <none>
hourly-backup-20210802100005   InProgress   0        0          2021-08-02 12:00:06 +0200 CEST   1h        default            <none>
hourly-backup-20210802090030   InProgress   0        0          2021-08-02 11:00:30 +0200 CEST   42m       default            <none>
test-backup                    InProgress   0        0          2021-07-28 20:58:34 +0200 CEST   24d       default            <none>
test-backup-1-5                Completed    0        0          2021-07-28 22:12:38 +0200 CEST   24d       default            <none>
test-backup-1-6                InProgress   0        0          2021-07-28 22:17:56 +0200 CEST   24d       default            <none>
weekly-backup-20210801230013   InProgress   0        0          2021-08-02 01:00:46 +0200 CEST   28d       default            <none>
weekly-backup-20210728170445   InProgress   0        0          2021-07-28 19:06:50 +0200 CEST   24d       default            <none>
weekly-backup-20210704230010   Completed    0        0          2021-07-05 01:01:17 +0200 CEST   14h       default            <none>
```

I see nothing  in the pod logs but i noticed that the pod restarted multiple times
```
$ kubectl get po
NAME                               READY   STATUS    RESTARTS   AGE
statsd-exporter-6c76d8fb66-q2rzn   2/2     Running   0          3d22h
velero-8548d7ff76-qx8f8            1/1     Running   99         3d22h    <<<<<<<<<<<<<<<<<<<<
```

and here are the previous pod logs : https://gist.github.com/yogeek/c5cec0bff1ecd5df10194dde07b2c9bf

**Comment by echel0n** [comment]
I can also see the Velero pod restarting

![image](https://user-images.githubusercontent.com/1128022/128039576-b90d0fb7-c5e2-4aed-9cb8-6a2089f45b69.png)

**Comment by echel0n** [comment]
After further investigation I was able to determine Velero was being terminated due to memory usage outside the resource limits, I've increased those and all is good.

Thanks!

**Comment by yogeek** [comment]
@echel0n Thanks for the update but on my side, I checked metrics in grafana and the memory usage does not seem to have reached the limit... (FYI : my velero memory requests/limit are currently set to  : 128Mo/256Mo...)
How did you check on your side please ?

**Comment by echel0n** [comment]
@yogeek as you stated, your pod shows that is has been restarted numerous times, this is a fairly good indication that it's been OOMKilled due to resource usage, you can simply just `kubectl describe po podname` and check if it's been terminated to confirm that.

However I use Botkube and Prometheus to track metrics and alerts so I got more information about it being killed due to the resource usage from Botkube, simply raise mem limits for both Velero and Restic to 1Gi should resolve it, I raised mine to 4Gi as mine was being killed at around 2Gi

**Comment by zubron** [comment]
I agree with @echel0n. Please run those commands to check why the pod is being restarted. As for the backups remaining in progress, this is a known issue with Velero where in progress backups are not updated/processed after pod restarts (see  https://github.com/vmware-tanzu/velero/issues/2877). You can also find instructions on how to update the resource limit in [our docs](https://velero.io/docs/v1.6/customize-installation/#customize-resource-requests-and-limits).

**Comment by wpjunior** [comment]
Hello, I am suffering by same issue, no pod restarts at all, could I watch the progress of a backup ?

**Comment by yogeek** [comment]
I have not been able to find the cause of the restarts but increasing the memory limit seems to solve the issue.
I have not seen any restart or backup error during the last week, since I increaded mem limit from 256Mi to 512Mi (which is apparently the [default](https://velero.io/docs/v1.6/customize-installation/#customize-resource-requests-and-limits) on in 1.6)
Thank you all for your help !

**Comment by yogeek** [comment]
After several weeks, this problem has not appeared anymore.
It was surely linked to the memory limit.
I think we can close this issue now.

**Comment by dirsigler** [comment]
So I just upgraded all Velero installations on our production Clusters to 1.7.0 and also updated to the latest GCP Plugin (1.3.0?).
While checking if the update succeeded I ran into the same issue as the issue reporter. Our biggest cluster setup had no proper backups and Velero only reported a InProgress state.

Doing a per-namespace backup seem to work, but there were no error logs or Events reported by the Container.

Upgrading the MemLimit of the Deployment to 512Mi solved the issue (thanks to @yogeek and @echel0n !).

There are now two "action points" I would love to be solved here:
- checking if we can have more logs or Container Events to better indicate when Velero runs into a Memory issue
- add a section in the troubleshooting documentation for this problem and resolution
